### PR TITLE
Update default headers azure & Add Token Provider Check

### PR DESCRIPTION
**Pull request by sarangan12**
This PR is to:

1.  Add a default user agent to header requests for Azure requests in the `@langchain/openai` library.
2. We need to check for instance name & deployment name when using the token provider

@achandmsft @deyaaeldeen @xirzec FYI.....

### CODE REVIEW

File: libs/langchain-openai/src/azure/chat_models.ts [hunk 1]
```diff
@@ -70,6 +71,14 @@ export class AzureChatOpenAI extends ChatOpenAI {
         delete params.baseURL;
       }
 
+      params.defaultHeaders = {
+        ...params.defaultHeaders,
+        "User-Agent":
+          params.defaultHeaders && params.defaultHeaders["User-Agent"]
```
Comment (sinedied) [inline libs/langchain-openai/src/azure/chat_models.ts:77]:
```suggestion
          params.defaultHeaders?.["User-Agent"]
```

File: libs/langchain-openai/src/azure/embeddings.ts [hunk 2]
```diff
@@ -64,6 +65,14 @@ export class AzureOpenAIEmbeddings extends OpenAIEmbeddings {
         delete params.baseURL;
       }
 
+      params.defaultHeaders = {
+        ...params.defaultHeaders,
+        "User-Agent":
+          params.defaultHeaders && params.defaultHeaders["User-Agent"]
```
Comment (sinedied) [inline libs/langchain-openai/src/azure/embeddings.ts:71]:
```suggestion
          params.defaultHeaders?.["User-Agent"]
```

File: libs/langchain-openai/src/azure/llms.ts [hunk 3]
```diff
@@ -66,6 +67,14 @@ export class AzureOpenAI extends OpenAI {
         delete params.baseURL;
       }
 
+      params.defaultHeaders = {
+        ...params.defaultHeaders,
+        "User-Agent":
+          params.defaultHeaders && params.defaultHeaders["User-Agent"]
```
Comment (sinedied) [inline libs/langchain-openai/src/azure/llms.ts:73]:
```suggestion
          params.defaultHeaders?.["User-Agent"]
```

**Comment by sinedied** [comment]
@jacoblee93 could you help us get this one merged? 🙏 

It fixes a blocking issue when the token provider, and sets the user agent in a similar way as we did in the previous Azure Open integration. Thank you!

**Comment by jacoblee93** [comment]
Sure, can cherry pick this one too if urgent.

**Pull request merged**